library/ssl_tls.c

```diff
@@ -6618,7 +6589,9 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
                 /* With DTLS, drop the packet (probably from last handshake) */
 #if defined(MBEDTLS_SSL_PROTO_DTLS)
                 if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
-                    return( MBEDTLS_ERR_SSL_WANT_READ );
+                {
+                    continue;
+                }
```

Comment:
@hanno-arm I see, in this case, do you think the sample applications should be modified? At the moment, they call `mbedtls_ssl_handshake`, if the error  is `WANT_READ` or `WANT_WRITE`. If now `WANT_READ` is not returned, will the applications work? Should they be modified not to check  `WANT_READ`?